clients/src/test/java/org/apache/kafka/common/requests/ApiVersionsResponseTest.java

```diff
@@ -166,7 +166,7 @@ public void testMetadataQuorumApisAreDisabled() {
         ApiVersionsResponse response = new ApiVersionsResponse.Builder().
             setThrottleTimeMs(AbstractResponse.DEFAULT_THROTTLE_TIME).
             setApiVersions(ApiVersionsResponse.filterApis(
-                ListenerType.ZK_BROKER,
+                ListenerType.BROKER,
```

Comment:
It appears that this test is no longer necessary, as it was designed to verify that these RPCs were not exposed through the zk broker. Or we can keep it but we should update the comment and test case name.

Reply:
Hmm, but do we have a test that verifies we don't expose controller apis on the broker listener? This seems to be doing that too.

Reply:
Although not sure why we removed one of the asserts in this test - I had missed that.

Reply:
> Although not sure why we removed one of the asserts in this test - I had missed that.

[DescribeQuorumRequest](https://github.com/apache/kafka/blob/trunk/clients/src/main/resources/common/message/DescribeQuorumRequest.json#L19) is exposed by kraft broker, so the assert `assertFalse(exposedApis.contains(ApiKeys.DESCRIBE_QUORUM));` will fail after we change the type from zk to kraft

> but do we have a test that verifies we don't expose controller apis on the broker listener? This seems to be doing that too.

yes, we can add that test but it should check all RPCs rather than only QUORUM-related RPCs.

Reply:
Yes, makes sense to have a test that checks all rpcs. Before we add it, we should check we don't have it already.